odoo/addons/base/models/ir_ui_view.py

```diff
@@ -410,7 +410,7 @@ def _check_xml(self):
         # Sanity checks: the view should not break anything upon rendering!
         # Any exception raised below will cause a transaction rollback.
         partial_validation = self.env.context.get('ir_ui_view_partial_validation')
-        self = self.with_context(validate_views=(self if partial_validation else True))
+        self = self.with_context(validate_view_ids=(self._ids if partial_validation else True))
```

Comment:
Is this assumed to be called with relatively few views so a tuple's fine to look up in?

Reply:
To my knowledge, views are almost never created in batch.
So yes, the average size of `validate_view_ids` must be very close to 1.